cpp/include/cudf/io/parquet.hpp

```diff
@@ -859,82 +836,139 @@ class chunked_parquet_writer_options_builder {
 };
 
 /**
- * @brief Forward declaration of anonymous chunked-writer state struct.
+ * @brief Merges multiple raw metadata blobs that were previously created by write_parquet
+ * into a single metadata blob
+ *
+ * @ingroup io_writers
+ *
+ * @param[in] metadata_list List of input file metadata
+ * @return A parquet-compatible blob that contains the data for all rowgroups in the list
  */
-struct pq_chunked_state;
+std::unique_ptr<std::vector<uint8_t>> merge_rowgroup_metadata(
+  const std::vector<std::unique_ptr<std::vector<uint8_t>>>& metadata_list);
+
+/**
+ * @brief parquet writer class to handle options and write.
+ *
+ * The following code snippet demonstrates how to write columns to a file:
+ * @code
+ *  ...
+ *  std::string filepath = "dataset.parquet";
+ *  cudf::io::parquet_writer_options options =
+ *  cudf::io::parquet_writer_options::builder(cudf::sink_info(filepath), table->view());
+ *  ...
+ *  cudf::io::parquet_writer writer(options);
+ *  writer.write()
+ * @endcode
+ */
+class parquet_writer {
```

Comment:
IMO we can remove this class and keep the function-based API for regular use case. AFAIK, Python API only uses the writer for chunked writes.